spring-kafka/src/main/java/org/springframework/kafka/support/converter/MessagingMessageConverter.java

```diff
@@ -66,13 +71,23 @@ public void setGenerateTimestamp(boolean generateTimestamp) {
 		this.generateTimestamp = generateTimestamp;
 	}
 
+	/**
+	 * Set the header mapper to map headers.
+	 * @param headerMapper the mapper.
+	 * @since 2.1
```

Comment:
Fine with me either way. We have postponed SI release anyway for some reason.
On the other hand this feature will be shipped only in the SCSt 2.0 anyway...

We might consider something like `1.3` Java 7 based and with the Kafka 0.11 but only with headers and admin client support.

We can definitely deffer transaction management to `2.1`.

Just some thought to come up with compromise...